scripts/experiments/tunnel_community/hidden_peer_discovery.py

```diff
@@ -28,8 +28,8 @@ def __init__(self, *args, **kwargs):
         self.register_task('_graceful_shutdown', self._graceful_shutdown, delay=EXPERIMENT_RUN_TIME)
 
     def _graceful_shutdown(self):
-        task = self.async_group.add_task(self.on_tribler_shutdown())
-        task.add_done_callback(lambda result: TinyTriblerService._graceful_shutdown(self))
+        if task := self.async_group.add_task(self.on_tribler_shutdown()):
+            task.add_done_callback(lambda result: TinyTriblerService._graceful_shutdown(self))
```

Comment:
I think this piece of code confirms what I'm saying. Here we have a code for **graceful shutdown**. That is, the code **should** be executed during the **shutdown**. With the design approach taken in this PR, this code can be totally and silently skipped. What kind of graceful shutdown is this if it can be completely ignored without any warning?